### PR TITLE
Support setting namespaces on gateway charts

### LINKED ISSUE #32655: Support setting namespaces on gateway charts
**Describe the feature request**

Kubernetes allows setting the namespaces when configuring pod antiaffinity, see https://v1-17.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.17/#podaffinityterm-v1-core

The gateways/istio-ingress chart should support setting the namespaces. See https://github.com/istio/istio/blob/master/manifests/charts/gateways/istio-ingress/templates/_affinity.tpl#L78

**Describe alternatives you've considered**

None.

[X] Docs
[X] Installation
[ ] Networking
[X] Performance and Scalability
[ ] Extensions and Telemetry
[ ] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

**Additional context**

### PR DESCRIPTION
**Pull request by shamsher31**
To fix https://github.com/istio/istio/issues/32655

### CODE REVIEW

File: manifests/charts/gateways/istio-egress/templates/_affinity.tpl [hunk 1]
```diff
@@ -70,6 +70,13 @@ nodeAffinity:
           {{- end }}
           {{- end }}
       topologyKey: {{ $item.topologyKey }}
+      {{- if $item.namespaces }}
+      namespaces:
+      {{- $ns := split "," $item.namespaces }}
+      {{- range $i, $n := $ns }}
+      - {{ $n | quote }}
+      {{- end }}
+      {{- end }}
```
Comment (morvencao) [inline manifests/charts/gateways/istio-egress/templates/_affinity.tpl:79]: Is the `namespaces` in affinity added after k8s 1.17?

Comment (shamsher31) [reply manifests/charts/gateways/istio-egress/templates/_affinity.tpl:79]: I think it was around from k8s 1.14.

**Review by morvencao — APPROVED** [review]

File: manifests/charts/gateways/istio-egress/templates/_affinity.tpl (hunk 1 above)
Comment (shamsher31) [reply manifests/charts/gateways/istio-egress/templates/_affinity.tpl:79]: Here is the ref:
https://v1-17.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.14/#podaffinityterm-v1-core

**Pull request merged**